src/app/ReadClient.h

```diff
@@ -164,6 +164,12 @@ class ReadClient : public Messaging::ExchangeDelegate
          * @param[in] apReadClient The read client object of the terminated read or subscribe interaction.
          */
         virtual void OnDone(ReadClient * apReadClient) = 0;
+
+        /**
+         * This function is invoked when the ReadClient was configured to auto re-subscribe and the ReadPrepareParams was
+         * moved into this client for management. This will have to be free'ed appropriately by the application.
```

Comment:
What's the contract here?  It only gets called if `SendAutoResubscribeRequest`, I assume?  Or does it get called for `SendSubscribeRequest` as well?  Or for other cases?

Does it get called if `SendAutoResubscribeRequest` fails?  Is it guaranteed to be called if `SendAutoResubscribeRequest` succeeds?  

Is this called before or after OnDone?

Reply:
OnDeallocatePaths is called in ReadClient's destructor when OnDone is called and delete ReadClient

Reply:
updated the comment, thanks